### ISSUE TITLE
JS沙箱问题

### ISSUE THREAD

**Issue by fitchsong** [umijs/qiankun#2418]
加载子应用后，在异步加载其他js文件，webpack  externals全局属性无法共享；

![image](https://user-images.githubusercontent.com/7778432/222089723-a6b69588-5aec-4510-9b10-c85c067395c6.png)
![image](https://user-images.githubusercontent.com/7778432/222089786-44412f16-e2c5-4281-875b-189c1abf2aa0.png)

源代码被两部分替换。请问externals 如果实现共享？

**Comment by gongshun** [comment]
子应用使用 externals 完全不影响呀，你是想让子应用使用主应用提供的 externals ？

**Comment by fitchsong** [comment]
> 子应用使用 externals 完全不影响呀，你是想让子应用使用主应用提供的 externals ？

不是。我子应用的部分js是异步加载的（直接操作dom 添加script标签）他与子应用的同步加载的js都被乾坤replace了；
造成的结果是，我在工程里externals的axios 添加的拦截器，在同步的js里添加的拦截器，在异步js里没有生效。

**Comment by NaLiTuo** [comment]
![image](https://user-images.githubusercontent.com/58428539/225202298-095ffaf6-7dad-41a7-94d6-1e528bc9e1d6.png)
这个api能否解决你的问题
